addon/templates/components/polaris-text-style.hbs

```diff
@@ -1,5 +1,26 @@
-{{#if hasBlock}}
-  {{yield}}
-{{else}}
-  {{text}}
+{{!-- NOTE: This is a bit hacky, but needed to support dynamically rendering different tag based on variation --}}
```

Comment:
not sure if there's a simple way to do this

Reply:
How about [`wrapper-element`](https://github.com/smile-io/ember-polaris/blob/master/addon/components/wrapper-element.js)? 🤔

Reply:
tried that, but it won't re-render if `elementTagName` changes

Reply:
True... maybe use the approach in [`polaris-list`](https://github.com/smile-io/ember-polaris/blob/master/addon/templates/components/polaris-list.hbs#L2) then?

Reply:
Still not perfect, but IMO an improvement 🤷‍♂️

Reply:
True... maybe use the approach in [`polaris-list`](https://github.com/smile-io/ember-polaris/blob/master/addon/templates/components/polaris-list.hbs#L2) then? Still not perfect but maybe an improvement 🤷‍♂️

Reply:
yeah, guess we can use that